packages/vuetify/src/components/VStepper/VStepper.tsx

```diff
@@ -0,0 +1,222 @@
+    const stepItems: any[] = []
+    const addSlotItem = (slotItem: any) => {
+      stepItems.push({
+        ...slotItem.props,
+        labelSlot: slotItem.children,
+        value: slotItem.props.step,
+        slot: contents?.filter(el => el.props.step === slotItem.props.step)[0]?.children,
+      }
+      )
+    }
+
+    const parseItems = () => {
+      // Reading slots and extracting the steps and step contents
+      addNodesInSlots('VStepperStep', slots, steps)
+      addNodesInSlots('VStepperContent', slots, contents)
+
+      /* Can be removed after approval
+      console.log('Slot Steps :')
+      console.log('===========================================')
+      console.log(steps)
+      console.log('Slot Contents :')
+      console.log('===========================================')
+      console.log(contents)
+      */
+
+      // Sorting both the property and slot items according to the step value
+      if (pitems.value.length > 0) pitems.value.sort((a: any, b: any) => (a?.step ?? 0 - b?.step ?? 0))
+      if (steps.length > 0) steps.sort((a: any, b: any) => (a?.props.step ?? 0 - b?.props.step ?? 0))
+
+      // Below Logic  is for checking both items (property/slots) and arrange them according to the priority
+      const pe = pitems.value.length !== 0
+      const se = steps.length !== 0
+
+      // console.log('Property Items exist : ' + pe + ', Slot Items exist : ' + se) // Can be removed after approval
+
+      if ((pe && props.dynamicFirst) || (!se && pe)) {
+        pitems.value.forEach(curItem => stepItems.push(curItem))
+        if (se) {
+          steps.forEach(curItem => {
+            if (pitems.value.filter((e: any) => (e.step ?? e.value ?? '') === curItem.props.step).length === 0) addSlotItem(curItem)
+          })
+        }
+      } else if ((se && !props.dynamicFirst) || (!pe && se)) {
+        steps.forEach(curItem => addSlotItem(curItem))
+        if (pe) {
+          pitems.value.forEach((curItem: any) => {
+            if (steps.filter((e: any) => (e.props.step ?? 1) === (curItem.step ?? curItem.value)).length === 0) stepItems.push(curItem)
+          })
+        }
+      }
+    }
+
+    parseItems()
+
+    const hasContent = contents.length > 0 || pitems.value.length > 0
+    stepItems.sort((a: any, b: any) => ((a?.step ?? 0) - (b?.step ?? 0))) // Sorting the final step items
+    // console.log(stepItems) // Can be removed after approval
+
+    provide(VStepperProvideSymbol, {
+      stackedLabels: toRef(props, 'stackedLabels'),
+      direction: toRef(props, 'direction'),
+    })
+
+    const classes = computed(() => {
+      const classList = [
+        'v-stepper',
+        `v-stepper--${props.direction}`,
+      ]
+      if (props.altLabels) classList.push('v-stepper--alt-labels')
+      if (props.flat) classList.push('v-stepper--flat')
+      if (props.nonLinear) classList.push('v-stepper--non-linear')
+      return classList
+    })
+    return () => {
+      if (props.direction === 'vertical') {
```

Comment:
This was one of the reasons for the hesitancy behind stepper. It's really difficult to do vertical and horizontal in the same component, especially without VNode parsing (which we are trying to avoid).

Reply:
I am not a fan of the parsing. But like you say v-stepper is a hard one because of the large difference in markup between the directions. Maybe we just make them different components with different markup? It would be harder for users to switch between the two directions at runtime, but I have to think that scenario is very uncommon.